### PR TITLE
[ML] Add suitable scoring functions for Text embedding models

### PR DESCRIPTION
**Pull request by davidkyle**
Adds a note about using a suitable scoring function to the compatible 3rd party text embedding models. 

Preview: https://stack-docs_2183.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-model-ref.html

I don't know how to link to the elasticsearch API docs from here, I've inserted the links but in a format which obviously won't work. 

I've edited the list of sentence transformer models to remove one model that is deprecated by the authors (https://huggingface.co/sentence-transformers/bert-base-nli-cls-token) and updated others to newer versions.

TODO: Cannot find any documentation about which scoring function the DPR models use

### CODE REVIEW

File: docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc [hunk 1]
```diff
@@ -65,20 +65,46 @@ refer to <<ml-nlp-overview>>.
...
 == Third party text embedding models
+Text Embedding models are designed to work with specific scoring functions
+for calculating the similarity between the embeddings they produce. 
+Examples of typical scoring functions are: `cosine`, dot product` and 
+`euclidean distance` (also known as `l2_norm`).
+
+The embeddings produced by these models should be indexed in Elasticsearch using the
+[dense vector field type](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html)
+with an appropriate [similarity function](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html#dense-vector-params)
+chosen for the model.
```
Comment (szabosteve) [inline docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc:76]:
```suggestion
The embeddings produced by these models should be indexed in Elasticsearch using the
[dense vector field type](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html)
with an appropriate {ref}/dense-vector.html#dense-vector-params[similarity function]
chosen for the model. 
```

File: docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc [hunk 2]
```diff
@@ -65,20 +65,46 @@ refer to <<ml-nlp-overview>>.
...
+`euclidean distance` (also known as `l2_norm`).
+
+The embeddings produced by these models should be indexed in Elasticsearch using the
+[dense vector field type](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html)
+with an appropriate [similarity function](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html#dense-vector-params)
+chosen for the model. 
+
+To find similar embeddings in Elasticsearch use the efficient Approximate k-nearest neighbor (kNN)
+search API with a text embedding as the query vector. 
+https://www.elastic.co/guide/en/elasticsearch/reference/master/knn-search.html#approximate-knn
```
Comment (szabosteve) [inline docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc:80]:
```suggestion
To find similar embeddings in {es} use the efficient {ref}/knn-search.html#approximate-knn[Approximate k-nearest neighbor (kNN)]
search API with a text embedding as the query vector. 
```

**Review by szabosteve — COMMENTED** [review]
The suggestions fix the formatting issues.

File: docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc [hunk 3]
```diff
@@ -65,20 +65,45 @@ refer to <<ml-nlp-overview>>.
 [discrete]
 [[ml-nlp-model-ref-text-embedding]]
 == Third party text embedding models
+Text Embedding models are designed to work with specific scoring functions
+for calculating the similarity between the embeddings they produce. 
+Examples of typical scoring functions are: `cosine`, dot product` and 
+`euclidean distance` (also known as `l2_norm`).
+
+The embeddings produced by these models should be indexed in Elasticsearch using the
```
Comment (davidkyle) [inline docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc:73]:
```suggestion
The embeddings produced by these models should be indexed in {es} using the
```

File: docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc [hunk 4]
```diff
@@ -65,20 +65,45 @@ refer to <<ml-nlp-overview>>.
...
+[dense vector field type](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html)
```
Comment (davidkyle) [inline docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc:74]:
```suggestion
{ref}/dense-vector.html[dense vector field type]
```

File: docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc [hunk 5]
```diff
@@ -65,20 +65,45 @@ refer to <<ml-nlp-overview>>.
...
+
+The embeddings produced by these models should be indexed in {es} using the
+{ref}/dense-vector.html[dense vector field type]
+with an appropriate {ref}/dense-vector.html#dense-vector-params[similarity function]
+chosen for the model. 
+
+To find similar embeddings in {es} use the efficient {ref}/knn-search.html#approximate-knn[Approximate k-nearest neighbor (kNN)]
+search API with a text embedding as the query vector. 
+
+Approximate KNN search uses the similarty function defined in
```
Comment (davidkyle) [inline docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc:81]:
```suggestion
Approximate kNN search uses the similarity function defined in 
```

File: docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc [hunk 6]
```diff
@@ -65,20 +65,45 @@ refer to <<ml-nlp-overview>>.
 [discrete]
 [[ml-nlp-model-ref-text-embedding]]
 == Third party text embedding models
+Text Embedding models are designed to work with specific scoring functions
```
Comment (szabosteve) [inline docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc:68]:
```suggestion

Text Embedding models are designed to work with specific scoring functions
```

File: docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc [hunk 7]
```diff
@@ -65,20 +65,45 @@ refer to <<ml-nlp-overview>>.
...
+The embeddings produced by these models should be indexed in {es} using the
+{ref}/dense-vector.html[dense vector field type]
+with an appropriate {ref}/dense-vector.html#dense-vector-params[similarity function]
+chosen for the model. 
+
+To find similar embeddings in {es} use the efficient {ref}/knn-search.html#approximate-knn[Approximate k-nearest neighbor (kNN)]
+search API with a text embedding as the query vector. 
+
+Approximate kNN search uses the similarity function defined in 
+the dense vector field mapping is used to calulate the relevence.
```
Comment (szabosteve) [inline docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc:82]:
```suggestion
the dense vector field mapping is used to calculate the relevance.
```

**Review by szabosteve — COMMENTED** [review]
Tiny suggestions.

**Pull request merged**